Automattic-Tracks-iOSTests/Event Logging/LogEncryptionTests.swift

```diff
@@ -0,0 +1,86 @@
+import XCTest
+import Sodium
+@testable import AutomatticTracks
+
+class LogEncryptionTests: XCTestCase {
+
+    private var keyPair: Box.KeyPair!
+    private var log: String!
+
+    override func setUp() {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        self.keyPair = Sodium().box.keyPair()!
```

Comment:
I just want to add about it being `stateless`. I am not a big fan of over-usage of properties in tests because I don't like that tests might modify things if they are run one after another. @shiki mentioned that XCode runs all tests in different instances, so that's not really a big issue in iOS, but I think it's good to keep in mind regardless.

For things that are stateful (mutable) I like having a helper method to create an instance with some parameters for modification. That way, instead of creating the property in the `setup`, a variable is created in the test itself and its usage and state are more obvious.

Not an incredibly important thing, but I have seen some test setups that made it very hard to track and debug in Android, so I am hoping at least the thought behind it would be helpful.

Reply:
Yep – this was done in 76c332f0bb990f24d39df000dfec0a879c7f9fb8.